coralproject/talk
# [CORL-1697]: Enable pre-moderation across single sites

## What does this PR do?

This adds the ability to configure pre-moderation for all comments across single sites for multi-site tenants. When this single site configuration is selected with one or more sites, those specific sites will have all of their comments pre-moderated. All other sites will have their comments post-moderated. This is for https://vmproduct.atlassian.net/browse/CORL-1697.

## What changes to the GraphQL/Database Schema does this PR introduce?

This adds `premoderateAllCommentsSites` to the settings. It also adds a `SINGLE_SITES_PRE` option to the `MODERATION_MODE`s.

I thought about adding `moderation` to `Site`s, but ultimately this seemed to make how we saved and retrieved the moderation information more complex. I was also thinking that it's starting to make sense to consolidate premoderation settings into a `PremoderationConfiguration`, with things like `premodLinksEnable`, `premoderateSuspectWords`, and `premoderateAllCommentsSites`. Or maybe a `ModerationConfiguration` that also includes the moderation mode? I didn't, however, make this change in the scope of this PR, but I'd be interested in others' thoughts on if I should, or if perhaps a new issue should be opened to address this.

## How do I test this PR?

- Test that the `SINGLE_SITES` option only shows up for multi-site tenants (`ON`/`OFF` should still be the options for single-site tenants)
- Test that `All sites` and `Off` options still pre-moderate/post-moderate all comments as expected
- Test that `Single sites` pre-moderates all comments for selected sites and not for others. Go to stream and add comments; see that they don't get added to the stream right away but go to pending moderation queue.
- Test that if `Single sites` is selected and no site is added, there is an error state that says at least one site must be selected
- Test that if a story is configured in the stream to pre-moderate all comments, this still works as expected, no matter which tenant option is selected
 
## How do we deploy this PR?
